### PR TITLE
feat: Add GraphQL Schema Visualisation

### PR DESCRIPTION
**Pull request by ghost**
This PR adds a visual introspection of our Graphql API.

New routes added:
 * Proxied public introspection results: https://api.1281.planx.pizza/introspect
 * Proxied admin introspection results: https://api.1281.planx.pizza/introspect-all
 * Public GraphQL graph visualisation: https://api.1281.planx.pizza/introspect/graph
 * Admin GraphQL graph visualisation: https://api.1281.planx.pizza/introspect-all/graph

(active duplicate of #1279 due to pizza issue.)

### CODE REVIEW

File: api.planx.uk/hasura/voyager.ts [hunk 1]
```diff
@@ -21,9 +21,12 @@ export default function graphQLVoyagerHandler(
     };
     if (adminKey) headers["x-hasura-admin-secret"] = adminKey;
 
-    const gqlUrl: string = process.env.HASURA_GRAPHQL_URL!;
+    const host: string = req.hostname;
+    const graphQLHost = host.replace(/^https:\/\/api/, "https://hasura");
+    const graphQLURL = `${graphQLHost}/v1/graphql`;
```
Comment (DafyddLlyr) [inline api.planx.uk/hasura/voyager.ts:26]: We have the env vars `API_URL_EXT` and `EDITOR_URL_EXT` which are the external URLs of these services, but we've not hit the requirement for a `HASURA_URL_EXT` yet.

Adding that might be a good long term solution here 👍

File: scripts/pullrequest/update.sh [hunk 2]
```diff
@@ -1,3 +1,6 @@
 #!/bin/bash
 
-DOCKER_BUILDKIT=1 docker-compose --env-file .env.prod -f docker-compose.yml -f docker-compose.pizza.yml up --build --force-recreate --remove-orphans -d
+DOCKER_BUILDKIT=1 docker-compose --env-file .env.prod \
+  -f docker-compose.yml \
+  -f docker-compose.pizza.yml \
+  up --build --renew-anon-volumes --force-recreate --remove-orphans -d
```
[the remark this answers is not in the document]
Comment (gunar) [reply scripts/pullrequest/update.sh:6]: I usually accomplish this preceding the `docker compose up` command with a `docker compose down --volumes` but thanks for sharing! I'll use that.

File: docker-compose.yml [hunk 3]
```diff
@@ -126,7 +126,6 @@ services:
     volumes:
       - "./api.planx.uk:/api"
       - "/api/node_modules"
-      - "/api/dist"
```
[the remark this answers is not in the document]
Comment (gunar) [reply docker-compose.yml:128]: Yeah I don't believe these should've been volumes.

File: api.planx.uk/server.ts [hunk 4]
```diff
@@ -352,8 +353,13 @@ app.get("/pay/:localAuthority/:paymentId", (req, res, next) => {
...
             const slack = SlackNotify(process.env.SLACK_WEBHOOK_URL!);
-            const getStatus = (state: Record<string, string>) => state.status + (state.message ? ` (${state.message})` : "")
-            const payMessage = `:coin: New GOV Pay payment *${govUkResponse.payment_id}* with status *${getStatus(govUkResponse.state)}* [${req.params.localAuthority}]`;
+            const getStatus = (state: Record<string, string>) =>
+              state.status + (state.message ? ` (${state.message})` : "");
+            const payMessage = `:coin: New GOV Pay payment *${
+              govUkResponse.payment_id
+            }* with status *${getStatus(govUkResponse.state)}* [${
+              req.params.localAuthority
+            }]`;
```
Comment (gunar) [inline api.planx.uk/server.ts:362]: This diff scares me. I'm assuming it's just prettification

**Review by gunar — COMMENTED** [review]
Took me a little while to understand what you meant to do in this PR but honestly I now love it!!

image


Once we get all CI steps passing this seems good to go.

**Pull request merged**